### PR TITLE
Add env variable to configure ImagePullPolicy for Deployments

### LINKED ISSUE #38: Missing configuration option for ImagePullPolicy on Deployments
Came across this while figuring out why the `image_pull_policy` environment variable, which is included the Helm chart, wasn't working.

Seems it hasn't been implemented yet so I've started working on a quick PR. Will post it tonight or tomorrow.

### PR DESCRIPTION
**Pull request by edvald**
First PR - and incidentally also my very first PR in Go!
Which is to say, please don't hesitate to point out if I made some noob mistake 😬 

I was a little unsure on how to do the input validation, maybe there are other more elegant/idiomatic ways to go about that?

Closes #38 (I hope)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -58,6 +67,14 @@ func main() {
 		functionNamespace = namespace
 	}
 
+	imagePullPolicy := corev1.PullAlways
+	if val, exists := os.LookupEnv("image_pull_policy"); exists {
+		if err = validation.In(pullPolicyOptions...).Validate(val); err != nil {
```
Comment (alexellis) [inline main.go]: Not sure about vendoring`validation.In`, we can just write some regular Go code to see if a string is in a slice or not.

Comment (edvald) [reply main.go]: Sure, makes sense, especially if you don't see more of this type of stuff propping up.

**Comment by alexellis** [comment]
Released in https://github.com/openfaas-incubator/openfaas-operator/releases/tag/0.8.3

@edvald please make a change to the helm chart in the faas-netes repo, if you need help ask in #faas-netes on Slack. There is already a flag for the imagePullPolicy for faas-netes, so you could follow the same pattern for this controller.

Alex

**Comment by edvald** [comment]
@alexellis - actually the variable was actually already there: https://github.com/openfaas/faas-netes/blob/e4a44772ce22596f05064b7bf3275409461a7ba9/chart/openfaas/templates/gateway-dep.yaml#L105

Unless I'm missing something else that needs to be added, I think it's good to go. I'll try it out and see if I bump into something.

**Comment by stefanprodan** [comment]
Yes I’ve added that some time ago but didn’t got time to do the implementation. Should work now.

**Pull request merged**